src/main/webapp/app/localci/build-queue/build-queue.component.html

```diff
@@ -660,3 +682,131 @@ <h3 id="build-queue-finished-heading" jhiTranslate="artemisApp.buildQueue.finish
+                        <div class="input-group-prepend form-item-prepend">
+                            <span class="input-group-text" jhiTranslate="artemisApp.buildQueue.filter.buildStartDate.from"></span>
+                        </div>
+                        <jhi-date-time-picker
+                            id="field_startDate"
+                            [shouldDisplayTimeZoneWarning]="false"
+                            [labelName]="''"
+                            [(ngModel)]="finishedBuildJobFilter.buildStartDateFilterFrom"
+                            [error]="!finishedBuildJobFilter.areDatesValid"
+                            (valueChange)="filterDateChanged()"
+                        />
+                    </div>
+                    <div class="input-group">
+                        <div class="input-group-append form-item-prepend">
+                            <span class="input-group-text" jhiTranslate="artemisApp.buildQueue.filter.buildStartDate.to"></span>
+                        </div>
+                        <jhi-date-time-picker
+                            id="field_endDate"
+                            [shouldDisplayTimeZoneWarning]="false"
+                            [labelName]="''"
+                            [(ngModel)]="finishedBuildJobFilter.buildStartDateFilterTo"
+                            [error]="!finishedBuildJobFilter.areDatesValid"
+                            (valueChange)="filterDateChanged()"
+                        />
+                        @if (!finishedBuildJobFilter.areDatesValid) {
+                            <span class="invalid-feedback"> {{ 'artemisApp.buildQueue.filter.buildStartDate.invalidDate' | artemisTranslate }}</span>
+                        }
+                    </div>
+                </div>
+            </div>
+
+            <div class="row finished-build-jobs-filter-border-bottom">
+                <div class="col d-flex flex-row pb-1">
+                    <h5 class="my-0">{{ 'artemisApp.buildQueue.filter.buildDuration.title' | artemisTranslate }}</h5>
+                </div>
+            </div>
+            <div class="row finished-build-jobs-filter-background-accent">
+                <div class="col container px-3 my-0 py-2">
+                    <div class="input-group">
+                        <span class="input-group-text" jhiTranslate="artemisApp.buildQueue.filter.buildDuration.lowerBound"></span>
+                        <input type="number" class="form-control" [(ngModel)]="finishedBuildJobFilter.buildDurationFilterLowerBound" (change)="filterDurationChanged()" />
+                        <span class="input-group-text" jhiTranslate="artemisApp.buildQueue.filter.buildDuration.upperBound"></span>
+                        <input type="number" class="form-control" [(ngModel)]="finishedBuildJobFilter.buildDurationFilterUpperBound" (change)="filterDurationChanged()" />
+                        @if (!finishedBuildJobFilter.areDurationFiltersValid) {
```

Comment:
This is also a function call in the template (`get ...()` syntax). Could we try to make it so the template does not rely on function calls?